### PR TITLE
Save QuestionnaireResponse as draft

### PR DESCRIPTION
**Pull request by joelfisch**

### CODE REVIEW

File: src/app/questionnaire-form-filler/questionnaire-form-filler.component.ts [hunk 1]
```diff
@@ -43,28 +49,41 @@ export class QuestionnaireFormFillerComponent implements OnInit {
...
               resourceType: 'QuestionnaireResponse',
-              input: response,
+              input: this.questionnaireResponse,
             })
             .then((data) => (this.extracted = data));
         }
       }
     }
-    this.questionnaireResponse = response;
+    return this.router.navigateByUrl('/');
```
Comment (sandrosc) [inline src/app/questionnaire-form-filler/questionnaire-form-filler.component.ts:74]: shouldn't we navigate back only when the submit action was finished?
And, for the other two functions as well, are the `return` statements necessary?

File: src/app/questionnaire-item/questionnaire-form-item-text/questionnaire-form-item-text.component.html [hunk 2]
```diff
@@ -27,6 +27,7 @@
   mat-raised-button
   color="primary"
   (click)="addItem()"
+  type="button"
```
Comment (sandrosc) [inline src/app/questionnaire-item/questionnaire-form-item-text/questionnaire-form-item-text.component.html:30]: These guys didn't anticipate the arrival of web apps when they decided that the default type is `submit` 😄

File: src/app/questionnaire-item/questionnaire-form/questionnaire-form.component.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<form (ngSubmit)="submitQuestionnaire.next()">
+<form (ngSubmit)="saveAsDraft.next()">
```
Comment (sandrosc) [inline src/app/questionnaire-item/questionnaire-form/questionnaire-form.component.html:1]: Shouldn't we use the type submit button to trigger the submit action?

Comment (sandrosc) [reply src/app/questionnaire-item/questionnaire-form/questionnaire-form.component.html:1]: IMO we could also solely rely on type button buttons.

Comment (joelfisch) [reply src/app/questionnaire-item/questionnaire-form/questionnaire-form.component.html:1]: I do not want to accidentally trigger a submit event and therefore I changed the default action of the submit. But yes, I think no submit at all would be an option as well.

File: src/app/questionnaire-item/questionnaire-form-item-text/questionnaire-form-item-text.component.html (hunk 2 above)
Comment (joelfisch) [reply src/app/questionnaire-item/questionnaire-form-item-text/questionnaire-form-item-text.component.html:30]: Yep 😅

File: src/app/questionnaire-form-filler/questionnaire-form-filler.component.ts (hunk 1 above)
Comment (joelfisch) [reply src/app/questionnaire-form-filler/questionnaire-form-filler.component.ts:74]: This is the case in the newest version. The return is not really necessary but it removes the warnings about not doing anything with the promise.

**Pull request merged**